lib/improver/tests/utilities/test_statistical_operations.py

```diff
@@ -0,0 +1,278 @@
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""
+Unit tests for the plugins and functions within statistical_operations.py
+"""
+
+import unittest
+import numpy as np
+
+import iris
+from iris.tests import IrisTest
+from iris.coords import DimCoord
+from iris.exceptions import CoordinateNotFoundError
+
+from improver.utilities.cube_checker import find_percentile_coordinate
+from improver.utilities.statistical_operations import \
+    ProbabilitiesFromPercentiles2D
+
+
+def set_up_percentiles_cube():
+    """ Set up 3D cube with percentiles of height """
+
+    test_data = np.full((5, 4, 4), -1)
+    for i in xrange(5):
+        test_data[i].fill(100*i + 200)
+
+    percentiles = DimCoord(np.linspace(0, 100, 5), long_name="percentiles",
+                           units="%")
+    grid_x = DimCoord(np.arange(4), standard_name="projection_x_coordinate",
+                      units="km")
+    grid_y = DimCoord(np.arange(4), standard_name="projection_y_coordinate",
+                      units="km")
+    test_cube = iris.cube.Cube(test_data, long_name="snow_level", units="m",
+                               dim_coords_and_dims=[(percentiles, 0),
+                                                    (grid_y, 1), (grid_x, 2)])
+    return test_cube
+
+
+def set_up_threshold_cube():
+    """
+    Set up 2D cube with "orography" data on which to threshold percentiles
+    """
+    test_data = 50*np.arange(16).reshape(4, 4)
+    grid_x = DimCoord(np.arange(4), standard_name="projection_x_coordinate",
+                      units="km")
+    grid_y = DimCoord(np.arange(4), standard_name="projection_y_coordinate",
+                      units="km")
+    test_cube = iris.cube.Cube(test_data, long_name="topography", units="m",
+                               dim_coords_and_dims=[(grid_y, 0), (grid_x, 1)])
+    return test_cube
+
+
+def set_reference_probabilities():
+    """
+    Define the array of probabilities correct for the percentile and threshold
+    cubes above.
+    """
+    reference_array = np.zeros(shape=(16,))
+    reference_array[:4] = 0.
+    reference_array[4:12] = np.linspace(0., 0.875, 8)
+    reference_array[12:] = 1.
+    return reference_array.reshape(4, 4)
+
+
+class Test__init__(IrisTest):
+    """ Test initialisation of the ProbabilitiesFromPercentiles2D class """
+    def setUp(self):
```

Comment:
I think you need tests to show that the `self.inverse_ordering` is set correctly. You probably need two tests, one for True and one for False

Reply:
✅